src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,4 +60,9 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
+    @Override
+    public int compareTo(Name o) {
+        return this.fullName.compareTo(o.toString());
```

Comment:
Try not to use `this` unless it is really needed, check 'Avoid unnecessary use of this with fields' in the coding standard.